backend/apps/events/tests.py

```diff
@@ -0,0 +1,937 @@
+                  id
+                  name
+                }
+              }
+            """,
+        )
+
+        # This validates the status code and if you get errors
+        self.assertResponseNoErrors(response)
+        # Fetching content of response
+        content = json.loads(response.content)
+
+        # There are two categories in the database
+        self.assertEqual(len(content["data"]["allCategories"]), 2)
+
+    def test_resolve_category(self):
+        response = self.query(
+            f"""
+            query {{
+                category(id: {self.first_category.id}) {{
+                  id
+                  name
+                }}
+              }}
+            """,
+        )
+
+        # This validates the status code and if you get errors
+        self.assertResponseNoErrors(response)
+        # Fetching content of response
+        content = json.loads(response.content)
+
+        # The correct category was fetched from the database
+        self.assertEqual(int(content["data"]["category"]["id"]), self.first_category.id)
+        self.assertEqual(content["data"]["category"]["name"], self.first_category.name)
+
+
+class EventsMutationsTestCase(EventsBaseTestCase):
+    """
+    Testing all mutations for events
+    """
+
+    def create_event(self, event, user=None):
+        query = f"""
+                mutation CreateEvent {{
+                    createEvent(
+                        eventData: {{
+                            title: \"{event.title}\",
+                            description: \"{event.description}\",
+                            startTime: \"{event.start_time.strftime("%Y-%m-%dT%H:%M:%S+00:00")}\",
+                            endTime:  \"{event.end_time.strftime("%Y-%m-%dT%H:%M:%S+00:00")}\",     
+                            organizationId: {event.organization.id},
+                            allowedGradeYears: {[int(grade) for grade in event.allowed_grade_years.split(",")]},
+                            contactEmail: \"{event.contact_email}\"
+                            }}
+                        ) {{
+                    event {{
+                        id
+                    }}
+                      ok
+                        }}
+                    }}
+                """
+        return self.query(query, user=user)
+
+    def create_attendable_event(self, event, attendable, slot_distribution, user=None):
+        query = ""
+
+        if event is not None and attendable is not None and slot_distribution is not None:
+            grade_years_string = self.stringify_grade_years(slot_distribution)
+            query = f"""
+                mutation CreateEvent {{
+                    createEvent(
+                        eventData: {{
+                            title: \"{event.title}\",
+                            description: \"{event.description}\",
+                            startTime: \"{event.start_time.strftime("%Y-%m-%dT%H:%M:%S+00:00")}\",
```

Comment:
These formatting strings could be extracted into a variable for better maintainability